src/core/evaluator.js

```diff
@@ -1640,19 +1650,22 @@ var PartialEvaluator = (function PartialEvaluatorClosure() {
                 xObjStateManager.transform(matrix);
               }
 
+              // enqueue textContent chunk before call for sub-form.
+              enqueueChunk();
               next(self.getTextContent({
                 stream: xobj,
                 task,
                 resources: xobj.dict.get('Resources') || resources,
                 stateManager: xObjStateManager,
                 normalizeWhitespace,
                 combineTextItems,
+                sink
               }).then(function (formTextContent) {
-                Util.appendToArray(textContent.items, formTextContent.items);
-                Util.extendObj(textContent.styles, formTextContent.styles);
+                // Util.appendToArray(textContent.items, formTextContent.items);
+                // Util.extendObj(textContent.styles, formTextContent.styles);
 
-                xobjsCache.key = name;
-                xobjsCache.texts = formTextContent;
+                // xobjsCache.key = name;
+                // xobjsCache.texts = formTextContent;
```

Comment:
> [...] so no need to cache.

Yes there is!
Since we don't want to potentially parse the *same* `XObject` over and over again, which is what your patch forces us to do, the current code caches its textContent and returns it immediately if it's already available; please see https://github.com/mozilla/pdf.js/blob/593dec1bb7aec1802abf8268137b0f7adab2ae32/src/core/evaluator.js#L1607-L1614.

This optimization should remain intact, even with the new code, since the point isn't whether we send the data immediately to the main thread. The point is that we don't want to parse any `XObject` more than once!